### PR TITLE
LPS-116924 Only users with edit permission for the layout and the asset should see metrics button.

### PR DESCRIPTION
**Pull request by cgoncas**
**Motivation**

Both content authors and marketing strategists are required to be involved in improving content performance and validating their strategy.
The content performance panel provides information on user engagement and visibility of online content, so anyone who can edit the page and content should have access to it. 

**Proposed solution**

Add permission checks to AnalyticsReportsProductNavigationControlMenuEntry, follow the same implementation we have in [EditDisplayPageMenuProductNavigationControlMenuEntry](https://github.com/liferay/liferay-portal/blob/d7986620aaf95241c1afc08d450666220bf24a9d/modules/apps/layout/layout-type-controller/layout-type-controller-display-page/src/main/java/com/liferay/layout/type/controller/display/page/internal/product/navigation/control/menu/EditDisplayPageMenuProductNavigationControlMenuEntry.java)

**How to test it**

Integration tests have been added to this pull.

### CODE REVIEW

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by cgoncas** [comment]
ci:forward

**Comment by cgoncas** [comment]
ci:stop

**Comment by cgoncas** [comment]
ci:test:stop

**Comment by cgoncas** [comment]
Reopening this pull, since it was accidentally forwarded to brian.

File: modules/dxp/apps/analytics/analytics-reports-test/src/testIntegration/java/com/liferay/analytics/reports/web/internal/product/navigation/control/menu/test/AnalyticsReportsProductNavigationControlMenuEntryTest.java [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+		_assetDisplayPageEntryLocalService;
+
+	@Inject
+	private CompanyLocalService _companyLocalService;
+
+	@DeleteAfterTestRun
+	private Group _group;
+
+	@Inject(filter = "component.name=*.JournalArticleInfoDisplayContributor")
+	private InfoDisplayContributor<FileEntry> _infoDisplayContributor;
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-test/src/testIntegration/java/com/liferay/analytics/reports/web/internal/product/navigation/control/menu/test/AnalyticsReportsProductNavigationControlMenuEntryTest.java:178]: Let's use a web content ;)

`private InfoDisplayContributor<JournalArticle> _infoDisplayContributor;`

**Comment by cgoncas** [comment]
Hey @darquesdev, this is exactly the same code you can find in [EditDisplayPageMenuProductNavigationControlMenuEntry](https://github.com/liferay/liferay-portal/blob/d7986620aaf95241c1afc08d450666220bf24a9d/modules/apps/layout/layout-type-controller/layout-type-controller-display-page/src/main/java/com/liferay/layout/type/controller/display/page/internal/product/navigation/control/menu/EditDisplayPageMenuProductNavigationControlMenuEntry.java), so I think that it is good enough for the Edit Display Page, it should be good enough for us.

File: modules/dxp/apps/analytics/analytics-reports-test/src/testIntegration/java/com/liferay/analytics/reports/web/internal/product/navigation/control/menu/test/AnalyticsReportsProductNavigationControlMenuEntryTest.java (hunk 1 above)
Comment (cgoncas) [reply modules/dxp/apps/analytics/analytics-reports-test/src/testIntegration/java/com/liferay/analytics/reports/web/internal/product/navigation/control/menu/test/AnalyticsReportsProductNavigationControlMenuEntryTest.java]: Fixed!!!

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
ci:test:relevant

**Comment by darquesdev** [comment]
That was a strong argument, actually I was advised against using that by the Echo team. So, go ahead, and we'll see in the future ;)

**Comment by cgoncas** [comment]
ci:forward

**Pull request closed without merging**